src/bindgen.rs

```diff
@@ -3,26 +3,57 @@
 use emoji;
 use error::Error;
 use progressbar::Step;
-use std::path::Path;
+use std::path::{Path, PathBuf};
 use std::process::Command;
+use which::which;
 use PBAR;
 
 /// Install the `wasm-bindgen` CLI with `cargo install`.
-pub fn cargo_install_wasm_bindgen(step: &Step) -> Result<(), Error> {
+pub fn cargo_install_wasm_bindgen(
+    path: &Path,
+    version: &str,
+    install_permitted: bool,
+    step: &Step,
+) -> Result<(), Error> {
+    // Determine if the `wasm-bindgen` dependency is already met for the given mode.
+    let dependency_met = if let Some(bindgen_path) = wasm_bindgen_path_str(path) {
+        wasm_bindgen_version_check(&bindgen_path, version).unwrap_or_else(|_| false)
+    } else {
+        false
+    };
```

Comment:
This is a little wonky right now, because the version check returns a result. As mentioned below, I think I can edit that so that it returns a plain boolean value, which should make this much nicer.